plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/providedservices/pde_monitor/routehandlers/GetAllPdeAlarms.java

```diff
@@ -0,0 +1,193 @@
+            systemNameParam, systemNameParam, severityParam, acknowledgedParam);
+
+        final QueryParamParser parser;
+
+        try {
+            parser = new QueryParamParser(null, acceptedParameters, request);
+        } catch (final ParseError error) {
+            logger.error("Encountered the following error(s) while parsing an HTTP request: " + error.getMessage());
+            response
+                .status(HttpStatus.BAD_REQUEST)
+                .body(ErrorMessage.of(error.getMessage()), CodecType.JSON);
+            return Future.success(response);
+        }
+
+        List<PdeAlarmDto> alarms = alarmManager.getAlarms();
+
+        final Optional<String> sortField = parser.getValue(sortFieldParam);
+        if (sortField.isPresent()) {
+            final String sortDirection = parser.getRequiredValue(directionParam);
+            final boolean ascending = ASC.equals(sortDirection);
+
+            switch (sortField.get()) {
+                case ID:
+                    PdeAlarm.sortById(alarms, ascending);
+                    break;
+                case RAISED_AT:
+                    PdeAlarm.sortByRaisedAt(alarms, ascending);
+                    break;
+                case UPDATED_AT:
+                    PdeAlarm.sortByUpdatedAt(alarms, ascending);
+                    break;
+                case CLEARED_AT:
+                    PdeAlarm.sortByClearedAt(alarms, ascending);
+                    break;
+                default:
+                    // We should never reach this case, since the sortField
+                    // param has been validated by the parser.
+                    throw new AssertionError("Encountered the invalid sort field '" + sortField + "'.");
+
+            }
+        }
+
+        final Optional<Integer> page = parser.getValue(pageParam);
+        if (page.isPresent()) {
+            final int itemsPerPage = parser.getRequiredValue(itemPerPageParam);
+
+            final int from = Math.min(page.get() * itemsPerPage, alarms.size());
+            final int to = Math.min(from + itemsPerPage, alarms.size());
+
+            alarms = alarms.subList(from, to);
+        }
+
+        final Optional<String> systemName = parser.getValue(systemNameParam);
+        if (systemName.isPresent()) {
+            PdeAlarm.filterBySystemName(alarms, systemName.get());
+        }
+
+        final Optional<String> severityValue = parser.getValue(severityParam);
+        if (severityValue.isPresent()) {
+            PdeAlarm.filterBySeverity(alarms, severityValue.get());
+        }
+
+        final Optional<Boolean> acknowledged = parser.getValue(acknowledgedParam);
+        if (acknowledged.isPresent()) {
+            PdeAlarm.filterAcknowledged(alarms, acknowledged.get());
+        }
+
+        PdeAlarmListDto result = new PdeAlarmListDto.Builder()
+            .data(alarms)
+            .count(alarms.size())
```

Comment:
count should be the size of alarms before paging

Reply:
Fixed.